### PR TITLE
Fix and refactor Load*MetadataSubscriber 

### LINKED ISSUE #5040: What is the purpose of Load(ORM/ODM)MetadataSubscriber?
Hi!

I'm using Sylius Resource Bundle to serve a rest API.

Simply, `LoadORMMetadataSubscriber` is breaking some the associations in my custom entities, specifically the `unsetAssociationMappings` method.

Before trying to fix the behaviour, I need to understand what it is for!
What is the purpose of these listeners?

Thanks

### PR DESCRIPTION
**Pull request by bendavies**
| Q               | A
| --------------- | ---
| Bug fix?      |  yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | fixes #5040
| License         | MIT

I've split the `Load*MetadataSubscriber` out into two different subscribers.
`*MappedSuperClassSubscriber`, which should only run on Sylius entities.
`*RepositoryClassSubscriber`, which should run on all entities.

Unfortunately there are no existing specs for mapped super class stuff.

### CODE REVIEW

**Comment by bendavies** [comment]
@pamil hello Sylius CI travis expert!
This is failing because I have a spec which is using Doctrine Mongo ODM classes, but suggested packages are not installed by the travis job. What do you think? We could install them with `--ignore-platform-reqs`?

**Comment by bendavies** [comment]
@pjedrzejewski ready for review.

**Comment by pamil** [comment]
@bendavies I'll try to get `mongodb` extension up and running on Travis and get rid of existing `--ignore-platform-reqs` for ResourceBundle, then we'll be able to add the packages requiring mongodb to `require-dev`.

File: src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+namespace Sylius\Bundle\ResourceBundle\EventListener;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use Sylius\Component\Resource\Metadata\RegistryInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php]: Is it true?

Comment (bendavies) [inline src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php]: well, i'm not sure of your convention. Symfony seemed to have a convention of putting Fabien as as the author of all files, regardless of who wrote the code.
I'll put myself if you want :)

**Comment by bendavies** [comment]
thanks @pamil !

File: src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php (hunk 1 above)
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php]: I think you should add yourself as an author :)

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php]: I am not sure if it is true for Symfony (probably no), but in Sylius we should definitely only mention the real authors of the file, not me everywhere. :)

**Comment by pamil** [comment]
@bendavies unfortunately there's no compatibile Doctrine MongoDB support for the PHP 7 driver, so it'll be more hacky solution :) We already have [Akeneo PhpSpecSkipExampleExtension](https://github.com/akeneo/PhpSpecSkipExampleExtension) included in main PhpSpec configuration, so adding `@require` annotations to specs requiring those classes will prevent running them on application level, whereas they will be run during ResourceBundle package tests (the only `--ignore-platform-reqs` package).

**Comment by bendavies** [comment]
@pamil done!

File: src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php (hunk 1 above)
Comment (bendavies) [inline src/Sylius/Bundle/ResourceBundle/EventListener/AbstractDoctrineSubscriber.php]: done :)

**Comment by pjedrzejewski** [comment]
Thank you Ben! One thing we should look at later is loading the ORM/ODM stuff only when doctrine ORM/ODM is actually installed in Symfony. Right now it just ignores the tags, but why put more services than needed. Also, ODM is not only MongoDB, so probably we should make the naming more specific. That's for the future I guess.

**Comment by bendavies** [comment]
you're welcome! yep i noticed that it could support mongo and phpcr. I guess lost of stuff needs  renaming to be Mongo specific.
Cheers!

**Pull request merged**